PhysicsTools/NanoAOD/plugins/GlobalVariablesTableProducer.cc

```diff
@@ -54,7 +54,7 @@ class GlobalVariablesTableProducer : public edm::stream::EDProducer<> {
                          "int", "float", "double", "bool", "candidatescalarsum", "candidatesize", "candidatesummass"));
     variable.add<edm::InputTag>("src")->setComment("input collection for the branch");
     variable.add<std::string>("doc")->setComment("few words description of the branch content");
-    variable.addOptional<int>("precision")->setComment("precision to store the information [NOT USED IN THE CODE]");
+    variable.addOptional<int>("precision")->setComment("precision to store the information");
```

Comment:
A  value to `precision_` is assigned in any case. Therefore, you can avoid the (disfavoured) usage of `existsAs(...)` and fully profit of the potentiality of the fillDescriptions method by defining here the default value of "-1" (i.e. the one to be assigned if the parameter is not present in the configuration) and remove the existsAs below at [L81](https://github.com/cms-sw/cmssw/pull/39955/files#diff-3031f3879df3ecd0ba8d198bc1db4d7a12c887db472454a91286def44d84cad5R81)
```suggestion
    variable.add<int>("precision",-1)->setComment("precision to store the information");
```

Reply:
I hear you. Rational for having "precision" it this way is that it's consistent (for good or bad) with the other nano modules. Also, with "exist as" let configurations where the parameter is not set to still function.